### PR TITLE
Make XmlSerializer work with NativeAOT

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Force the reflection-only code paths that don't Reflection.Emit.

### CODE REVIEW

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs [hunk 1]
```diff
@@ -199,6 +199,12 @@ public XmlSerializer(Type type, string? defaultNamespace)
                 _primitiveType = type;
                 return;
             }
+
+            if (Mode == SerializationMode.ReflectionOnly)
+            {
+                return;
```
Comment (jkotas) [inline src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs:205]: Do we need to also execute `GetTopLevelMapping` at the very bottom of the method in ReflectionOnly mode?

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs [hunk 2]
```diff
@@ -199,6 +199,12 @@ public XmlSerializer(Type type, string? defaultNamespace)
...
+            }
+
             _tempAssembly = s_cache[defaultNamespace, type];
             if (_tempAssembly == null)
```
Comment (jkotas) [inline src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs:209]: Do we need to fix ` XmlSerializer(XmlTypeMapping xmlTypeMapping)` constructor in the same way?

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs (hunk 1 above)
Comment (MichalStrehovsky) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs:205]: No, I think this was all ifdeffed out:

https://github.com/dotnet/corefx/blob/e31a2b7f3bdf985f56798df530daa4c06fd80376/src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs#L266-L308

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs (hunk 2 above)
Comment (MichalStrehovsky) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs:209]: Welp, I haven't done a full audit of what were all of the deleted ifdefs doing. This is more of an emergency fix to see what else the partner team is going to hit. We could file an issue to do a full audit.

**Review by jkotas — APPROVED** [review]

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs (hunk 2 above)
Comment (MichalStrehovsky) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs:209]: https://github.com/dotnet/runtimelab/issues/596

**Pull request merged**